@here/harp-mapview-decoder/lib/TileDataSource.ts

```diff
@@ -164,6 +164,10 @@ export class TileDataSource<TileType extends Tile> extends DataSource {
         this.decoder.dispose();
     }
 
+    clear() {
```

Comment:
@robertoraggi, I like @zbigg 's idea of just having the cache clearing in the `setStyleSet` method.

In the case you mentioned, the `clear()` method has a clear (forgive the pun) correlation to the `add()` method, however with the TileDataSource, it isn't apparent when `clear()` should be called, because the user of the class can't (directly) add anything to it.